core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -534,13 +516,18 @@ object Doc {
    */
   private[paiges] case object Empty extends Doc
 
+  /**
+    * Render 'default' except when flattened.
+    * Invariant: isEmpty(FlatAlt(a)) <=> isEmpty(FlatAlt(b))
```

Comment:
do you mean `isEmpty(a) == isEmpty(b)`?

Reply:
Ug, yes sorry.  I wanted to drop the rhs when testing for empty.

Reply:
`isEmpty` seems to be only used to check the separator is empty in, e.g. `intercalate`.  In my opinion this should not be exposed in the API.  The reason is that I think of the library as being used as follows:

- Put combinators together
- Pick a column width
- Run render

I don't expect the user to be branching on whether a `Doc` is empty, or if a `Doc` can be flattened or not.  This seems to be breaking an abstraction barrier in my way of thinking about the problem.  Is this not other's view of the usage purpose of the library?  If users are doing this branching, perhaps exposing `flatAlt` would allow them to get the behavior they want without needing to hack on the `Doc`s in a non-declarative fashion.